### PR TITLE
Refactor settings

### PR DESCRIPTION
**Pull request by Koenkk**
https://github.com/Koenkk/zigbee2mqtt.io/pull/1097

### CODE REVIEW

**Comment by sjorge** [comment]
Just something that popped into my head while skimming over the PR (which looks fine to me), can we somehow add a hook to the extension framework that would then be called inside the settings.ts to extend the schema.

That way everything is inside the homeassistent extension, and other extensions could use those hooks to expose there own options in a way that allows z2m / the frontend to validate it properly.

**Comment by Koenkk** [comment]
@sjorge I believe you can already do this by modifying `settings.schema` in the constructor of your extension

**Comment by nurikk** [comment]
It would be nice to introduce settings version property, using it we can implement seamless settings migrations.

**Comment by Koenkk** [comment]
@nurikk good idea, where would we store this property?

**Comment by nurikk** [comment]
> @nurikk good idea, where would we store this property?

Somewhere in root level of `configuration.yaml`? Same way as we do in `docker-compose.yml`

**Comment by Koenkk** [comment]
@nurikk not sure if that would be a bit overkill, I don't think the settings will change much in the feature, we can do simple migrations in e.g. https://github.com/Koenkk/zigbee2mqtt/pull/10437/files#diff-37c60e6d6e159be28d507ce5ede419ef5cb81effa6bda5641f58fbef7e255d73R162

**Comment by sjorge** [comment]
@Koenkk don't forget to mark this as a breaking change for people who have user extensions.

entity.settings got renamed to entity.options, resulting in my user extensions not working or worse throwing exceptions because entity.settings is undefined.

**Comment by Koenkk** [comment]
@sjorge thanks, I will add it to the release notes.

File: lib/extension/availability.ts [hunk 1]
```diff
@@ -13,17 +13,13 @@ export default class Availability extends Extension {
...
 
         const key = this.isActiveDevice(device) ? 'active' : 'passive';
-        const availabilitySettings = settings.get().availability;
-        if (typeof availabilitySettings === 'object' && availabilitySettings[key]?.timeout != null) {
-            return utils.minutes(availabilitySettings[key]?.timeout);
-        }
-
-        return key === 'active' ? utils.minutes(10) : utils.hours(25);
+        const value = settings.get().availability[key]?.timeout;
+        return key === 'active' ? utils.minutes(value) : utils.hours(value);
```
Comment (joshuaspence) [inline lib/extension/availability.ts:22]: I'm not sure if this was intentional or not but `availability.passive.timeout` is now specified in hours whereas the help text in the UI says "Time after which an passive device will be marked as offline in minutes"

Comment (Koenkk) [reply lib/extension/availability.ts:22]: whoops, nice catch. fixed it!

**Pull request merged**